lib/navigation/model/route_paths.dart

```diff
@@ -0,0 +1,347 @@
+  /// from here [lib/pages/portal/view/website_view.dart:254] is changed
+  /// OR remove [category] attribute which means that the usage from
+  /// [lib/pages/portal/view/portal_page.dart:358] will always redirect to
+  /// `category = [WebsiteCategory.learning]', which means a button on it's parent page
+  /// will have no feedback.
+  /// OR rename [category] to 'defaultCategory' :bigbrain:
+  @override
+  Widget get page {
+    return ChangeNotifierProxyProvider<AuthProvider, FilterProvider>(
+      create: (BuildContext context) {
+        return Platform.environment.containsKey('FLUTTER_TEST')
+            ? Provider.of<FilterProvider>(context)
+            : FilterProvider();
+      },
+      update: (context, authProvider, filterProvider) {
+        return filterProvider..updateAuth(authProvider);
+      },
+      child: WebsiteView(
+        website: Website(
+            relevance: null,
+            id: null,
+            isPrivate: true,
+            link: '',
+            category: _websiteCategory),
+      ),
+    );
+  }
+}
+
+class ClassesFeedbackPath extends RoutePath {
+  ClassesFeedbackPath() : super(ClassFeedbackChecklist.routeName);
+
+  @override
+  Widget get page => const ClassFeedbackChecklist();
+}
+
+class ClassFeedbackViewPath extends RoutePath {
+  ClassFeedbackViewPath(this.id) : super(ClassFeedbackView.routeName);
+
+  final String id;
+
+  // TODO(RazvanRotaru): Retrieve [ClassHeader] for [ClassFeedbackView] by id
+  @override
+  Widget get page => const ClassFeedbackView();
+}
+
+class ClassViewPath extends RoutePath {
+  ClassViewPath(this.id) : super(ClassView.routeName);
+
+  final String id;
+
+  // TODO(RazvanRotaru): retrieve [ClassHeader] for [ClassView] by id
+  @override
+  Widget get page => const ClassView();
+}
+
+class ClassesPagePath extends RoutePath {
+  ClassesPagePath() : super(ClassesPage.routeName);
+
+  @override
+  Widget get page => const ClassesPage();
+}
+
+class AddClassesPath extends RoutePath {
+  AddClassesPath() : super(AddClassesPage.routeName);
+
+  @override
+  Widget get page => const AddClassesPage();
+}
+
+class EventViewPath extends RoutePath {
+  EventViewPath(this.id) : super('${EventView.routeName}?id=$id');
+
+  final String id;
+
+  // TODO(RazvanRotaru): retrieve [Event] for [EventView] by id
+  @override
+  Widget get page => EventView();
+}
+
+class AddEventPath extends RoutePath {
+  AddEventPath() : super(AddEventView.routeName);
+
+  @override
+  Widget get page => const AddEventView();
+}
+
+class UnknownPath extends RoutePath {
+  UnknownPath() : super('');
+
+  // TODO(RazvanRotaru): Use [ErrorPage] instead of plain [Text]
+  @override
+  Widget get page => const Text('404! Unknown location');
+}
+
+// TODO(RazvanRotaru): auto-generate PathFactory
+class PathFactory {
+  static RoutePath from(Uri uri) {
+    if (uri.pathSegments.isEmpty) {
+      return RootPath();
+    }
+
+    // final String path = uri.path;
```

Comment:
Remove this?

Reply:
I believe I did it here https://github.com/student-hub/acs-upb-mobile/pull/265/commits/e429fbb2862a3256725bd4141000f0be5f097dc2 , anyway, it's gone